include/esp_memory_resource.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <memory_resource>
+#include <utility>
+
+#include "esp_heap_caps.h"
+
+namespace idf::memory {
+
+enum class capabilities : uint32_t {
+    not_specified = MALLOC_CAP_DEFAULT,
+    adressable_8bit = MALLOC_CAP_8BIT,
+    adressable_32bit = MALLOC_CAP_32BIT,
+    dma = MALLOC_CAP_DMA,
+    executable = MALLOC_CAP_EXEC,
+    from_spiram = MALLOC_CAP_SPIRAM,
+    internal = MALLOC_CAP_INTERNAL,
+    retention_dma = MALLOC_CAP_RETENTION,
+    rtc = MALLOC_CAP_RTCRAM
+};
+
+constexpr capabilities operator|(const capabilities lhs, const capabilities rhs)
+{
+    return capabilities{std::to_underlying(lhs) | std::to_underlying(rhs)};
+}
+
+class resource : public std::pmr::memory_resource {
```

Comment:
For my taste, a simple comment what this class does and guiding users to the documentation of PMR would be helpful.